programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -1223,6 +1268,63 @@ fn update_callee_account(
     Ok(())
 }
 
+fn update_caller_account_perms(
+    memory_mapping: &MemoryMapping,
+    caller_account: &CallerAccount,
+    callee_account: &BorrowedAccount<'_>,
+    is_loader_deprecated: bool,
+) -> Result<(), Error> {
+    let CallerAccount {
+        original_data_len,
+        vm_data_addr,
+        ..
+    } = caller_account;
+
+    let data_region = account_data_region(memory_mapping, *vm_data_addr, *original_data_len)?;
+    if let Some(region) = data_region {
+        match (
```

Comment:
Is the change / edge detection really useful here? Why not always set to the new state?

Reply:
I think you commented on an old commit, I'm going to resolve